zproject/backends.py

```diff
@@ -1200,6 +1200,13 @@ def social_associate_user_helper(backend: BaseAuth, return_data: Dict[str, Any],
                 # second time, we won't have any name data.  We handle
                 # this by setting full_name to be the empty string.
                 full_name = ""
+
+                # Python social auth code changes first_name and last_name
+                # to `None` when no name is returned by apple. Due to that
+                # retun_data["full_name"] becomes "None None" below. To
+                # avoid that we also make first_name and last_name empty strigs.
+                first_name = ""
+                last_name = ""
```

Comment:
we should fix the logic below to handle `None` values in a proper way rather than putting a hack in here

Reply:
So, we'll have something like this?
```diff
diff --git a/zproject/backends.py b/zproject/backends.py
index 6fc3491f7d..537c82e9e8 100644
--- a/zproject/backends.py
+++ b/zproject/backends.py
@@ -1210,6 +1210,10 @@ def social_associate_user_helper(backend: BaseAuth, return_data: Dict[str, Any],
     if full_name:
         return_data["full_name"] = full_name
     else:
+        if first_name is None:
+            first_name = ''
+        if last_name is None:
+            last_name = ''
         # In SAML authentication, the IdP may support only sending
         # the first and last name as separate attributes - in that case
         # we construct the full name from them.
~
```

No backend gives `None` for first or last names. So, since it's apple specific, I added those there and I felt like it's clear to convey that only apple auth backend gives it them as `None` the current way.